autogpt/plugins.py

```diff
@@ -18,6 +18,29 @@
 from autogpt.models.base_open_ai_plugin import BaseOpenAIPlugin
 
 
+def install_plugin_dependencies(zip_path: str, debug: bool = False) -> None:
+    """
+    Inspect a zipfile for a requirements file and install if exists. Only
+    searches at top level for now.
+
+    Args:
+        zip_path (str): Path to the zipfile.
+        debug (bool, optional): Enable debug logging. Defaults to False.
+
+    Returns:
+        None
+    """
+    try:
+        with zipfile.ZipFile(zip_path, "r") as zfile:
+            basedir = zfile.namelist()[0]
+            basereqs = os.path.join(basedir, 'requirements.txt')
+            extracted = zfile.extract(basereqs)
+            os.system(f"pip install -r {extracted}")
```

Comment:
This is a potential pain point. Can you get the running python exec and use it with -m pip …

Reply:
Would you rather just have this all in `check_requirements.py`?

Reply:
https://github.com/Significant-Gravitas/Auto-GPT/pull/3068/commits/576f9a125e0afe3f4331d02a7d25001b017f561f

If this is better, I can revert the previous changes ^